hare/refcounttracker.go

```diff
@@ -1,7 +1,5 @@
 package hare
 
-type objectId uint64
```

Comment:
If you pay close attention there was actually no use of ObjectID. I used it before when that was the identifier of objects. 
Since there are no generics in GO, I find adding ObjectID a bit unneeded.
This PR cleans up the code, there is no intention to change code segments unless it is cleaner. Since this point is debatable I went with the simplest one and preferred removing code than adding code.